robot/client/client.go

```diff
@@ -1078,3 +1083,54 @@ func (rc *RobotClient) Version(ctx context.Context) (robot.VersionResponse, erro
 
 	return mVersion, nil
 }
+
+func addVersionMetadataToContext(ctx context.Context) context.Context {
+	info, _ := debug.ReadBuildInfo()
+	deps := make(map[string]*debug.Module, len(info.Deps))
+	for _, dep := range info.Deps {
+		deps[dep.Path] = dep
+	}
+	var apiVersion string
+	if dep, ok := deps["go.viam.com/api"]; ok {
```

Comment:
Also, is it possible that `apiVersion` never gets set? Say, if someone were to build locally and point to a local `api` repo? Should we have a catch-all in that case?

Reply:
(See above comment)
Originally there was a character, but it was removed since I learned the standard was to leave an empty string in Go. I'll do the same thing I do for `version`.

Reply:
For the sake of consistency, we should do the same as other SDK when the apiVersion is empty. I believe both the CLI and the Robot api return `?` when apiVersion is empty.

Reply:
I think there's some slight inconsistency among other SDKs. Namely: for python, typescript, flutter, we always build based on a particular commit with a particular tag, so we can always pass that. For C++, we support users building with a local version of the `api` repo but no modifications to the SDK itself, meaning the `api` could technically be anything. In this case, instead of the `vX.Y.Z` tag, we pass `unknown (local build)` or something to that effect.

Reply:
So what should we return here? `unknown`, or `?`

Reply:
If we use the Version(), then apparently it returns `dev-unknown`, which I think is good